apache/incubator-zeppelin
# [ZEPPELIN-183] Interpreter process per note

### What is this PR for?

This PR give an option to create interpreter process per notebook.

### What type of PR is it?
Feature

### Todos
* [x] - Interpreter process per note
* [x] - Interpreter menu gui change
* [x] - doc update

### What is the Jira issue?
https://issues.apache.org/jira/browse/ZEPPELIN-183

### How should this be tested?
Change interpreter setting option to 'process' and then create two notebooks.
Try to use both notebook with the same interpreter setting.
You'll see each notebook creates different interpreter process.

For example spark interpreter, if you run 
```
%spark
sc
```
in two notebooks, then you'll see different object id.


### Screenshots (if appropriate)

Interpreter setting option gui looks like

![image](https://cloud.githubusercontent.com/assets/1540981/14034874/cd52f432-f1e4-11e5-8ba6-335551ae8ec4.png)

If you hover the pointer on each option, you'll see tooltip help about each mode.

### Questions:
* Does the licenses files need update? no
* Is there breaking changes for older versions? no
* Does this needs documentation? yes
